packages/foam-core/src/janitor/index.ts

```diff
@@ -0,0 +1,80 @@
+import { Position } from 'unist';
+import dashify from 'dashify';
+import { Note, NoteGraph } from '../index';
+import {
+  createMarkdownReferences,
+  stringifyMarkdownLinkReferenceDefinition,
+} from '../markdown-provider';
+import { getHeadingFromFileName } from '../utils'
+
+export interface TextEdit {
+  range: Position;
+  newText: string;
+}
+
+export const generateLinkReferences = (note: Note, ng: NoteGraph): TextEdit | null => {
+  const newReferences = createMarkdownReferences(ng, note.id).map(
+    stringifyMarkdownLinkReferenceDefinition
+  ).join('\n');
+
+  if (note.definitions.length === 0) {
+    if (newReferences.length === 0) {
+      return null;
+    }
+
+    // @todo: how to abstract new line?
+    const padding = note.end.column === 1 ? '\n' : '\n\n';
```

Comment:
Note @anku255 @chirag-singhal. We still need to figure out how/where to normalise newlines. Usually newline format depends on platform but in some cases (e.g. when using git), people's git configs can check out linux new lines on windows and vice versa -- not sure what the best way is.

Reply:
Will look into it.